src/lib/uptime_service/uptime_service.ml

```diff
@@ -0,0 +1,326 @@
+  in
+  let json =
+    (* JSON structure in issue 9110 *)
+    `Assoc
+      [ ("data", block_data_json)
+      ; ("signature", Signature.to_yojson signature)
+      ; ("submitter", Public_key.to_yojson submitter_keypair.public_key) ]
+  in
+  let headers = Cohttp.Header.of_list [("Content-Type", "application/json")] in
+  match%map
+    Monitor.try_with ~here:[%here] ~extract_exn:true (fun () ->
+        match%map
+          Cohttp_async.Client.post ~headers
+            ~body:(Yojson.Safe.to_string json |> Cohttp_async.Body.of_string)
+            url
+        with
+        | {status; _}, _body ->
+            if Cohttp.Code.code_of_status status = 200 then
+              [%log info]
+                "Sent block with state hash $state_hash to uptime server at \
+                 URL $url"
+                ~metadata:
+                  [ ("state_hash", State_hash.to_yojson state_hash)
+                  ; ( "includes_snark_work"
+                    , `Bool (Option.is_some block_data.snark_work) )
+                  ; ("url", `String (Uri.to_string url)) ]
+            else
+              [%log error]
+                "Failure when sending block with state hash $state_hash to \
+                 uptime server at URL $url"
+                ~metadata:
+                  [ ("state_hash", State_hash.to_yojson state_hash)
+                  ; ("url", `String (Uri.to_string url))
+                  ; ("http_code", `Int (Cohttp.Code.code_of_status status))
+                  ; ( "http_error"
+                    , `String (Cohttp.Code.string_of_status status) ) ] )
+  with
+  | Ok () ->
+      ()
+  | Error exn ->
+      [%log error]
+        "Error when sending block with state hash $state_hash to uptime \
+         server at URL $url"
+        ~metadata:
+          [ ("state_hash", State_hash.to_yojson state_hash)
+          ; ("url", `String (Uri.to_string url))
+          ; ("error", `String (Exn.to_string exn)) ]
+
+let block_base64_of_breadcrumb ~logger breadcrumb =
+  let external_transition = external_transition_of_breadcrumb breadcrumb in
+  let block_len =
+    External_transition.Stable.Latest.bin_write_t block_buf ~pos:0
+      external_transition
+  in
+  let block_string = String.init block_len ~f:(fun ndx -> block_buf.{ndx}) in
+  let block_base64_result = Base64.encode block_string in
+  match block_base64_result with
+  | Ok block_base64 ->
+      Some block_base64
+  | Error (`Msg err) ->
+      [%log error] "Could not Base64-encode block with state hash $state_hash"
+        ~metadata:
+          [ ( "state_hash"
+            , State_hash.to_yojson
+              @@ External_transition.state_hash external_transition )
+          ; ("error", `String err) ] ;
+      None
+
+let send_produced_block_at ~logger ~url ~transition_frontier ~peer_id
+    ~(submitter_keypair : Keypair.t) tm =
+  (* give block production some time *)
+  let%bind () = at (Time.add tm Time.Span.minute) in
```

Comment:
I don't know what the average block production time is in mainnet but our nodes on devnet take about a minute and a half to produce blocks. Should we make this 2 mins?